golang/thrift/inbound.go

```diff
@@ -0,0 +1,297 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "fmt"
+import "strings"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+
+// NewTChannelInboundProtocol creates a TChannelInboundProtocol
+func NewTChannelInboundProtocol(call *tchannel.InboundCall) *TChannelInboundProtocol {
+	return &TChannelInboundProtocol{
+		call: call,
+	}
+}
+
+//
+// TChannelInboundProtocol is a thrift.TProtocol implementation for
+// inbound (i.e., server side) tchannel calls. It serves as the adaptor
+// between the generated thrift server side code (thrift.TProcessor)
+// and tchannel (tchannel.InboundCall).
+//
+// Incoming and outgoing data is buffered in memory buffers and the
+// actual parsing of the data is delegated to thrift.TBinaryProtocol.
+//
+type TChannelInboundProtocol struct {
+	call *tchannel.InboundCall
+
+	writer      *thrift.TBinaryProtocol
+	writeBuffer *MemoryBufferTransport
+
+	reader     *thrift.TBinaryProtocol
+	readBuffer *MemoryBufferTransport
+}
+
+// WriteMessageBegin creates the writer buffer and a TBinaryProtocol
+// writer and delegates the write to the writer
+func (p *TChannelInboundProtocol) WriteMessageBegin(name string, typeId thrift.TMessageType, seqId int32) error {
+	p.writeBuffer = NewMemoryBufferTransport()
+	p.writer = thrift.NewTBinaryProtocol(p.writeBuffer, false, false)
+	return p.writer.WriteMessageBegin(name, typeId, seqId)
```

Comment:
This doesn't match what Node and Python are doing. This information is not written to the payload.